src/sys/socket/addr.rs

```diff
@@ -515,15 +515,44 @@ impl fmt::Display for Ipv6Addr {
 }
 
 /// A wrapper around `sockaddr_un`.
-///
-/// This also tracks the length of `sun_path` address (excluding
-/// a terminating null), because it may not be null-terminated.  For example,
-/// unconnected and Linux abstract sockets are never null-terminated, and POSIX
-/// does not require that `sun_len` include the terminating null even for normal
-/// sockets.  Note that the actual sockaddr length is greater by
-/// `offset_of!(libc::sockaddr_un, sun_path)`
 #[derive(Clone, Copy, Debug)]
-pub struct UnixAddr(pub libc::sockaddr_un, pub usize);
+pub struct UnixAddr {
+    // INVARIANT: sun+sun_len is valid as defined by docs for from_raw_parts
+    sun: libc::sockaddr_un,
+    sun_len: usize
```

Comment:
This is a pretty confusing name, since it's not the same as `sun.sun_len`.

Reply:
Ah, I didn't realize that there was a `sun.sun_len` in BSD, I see that now. What would you recommend? `addrlen` would be the name for the length including the `sa_family`, so maybe `path_len` or `buflen`?

Reply:
What exactly will be the difference between this field and `sun_len` ( on platforms where it exists)?  pathlen would probably be fine.